include/TriangleNodesProvisioner.hpp

```diff
@@ -0,0 +1,290 @@
+
+      const MeshManager * Mesh2D;
+      
+      std::unique_ptr<Nodes1DProvisioner> Nodes1D;
+	    JacobiBuilders Jacobi;
+      VandermondeBuilders Vandermonde;
+      DirectSolver LinSolver;
+
+  public:
+      static const index_type NumFaces;
+      static const real_type NodeTol;
+    /**
+     * Constructor.
+     * @param[in] NOrder Order of the approximating polynomials.
+     * @param[in] NumElements The number of elements.
+     * @param[in] MeshManager The MeshManager storing the 2D mesh.
+     * @note Assumes uniform elements.
+     */
+    TriangleNodesProvisioner(index_type _NOrder, index_type _NumElements, const MeshManager * _MeshManager);
+
+    /**
+     * Copy constructor (deleted).
+     */
+    TriangleNodesProvisioner(const TriangleNodesProvisioner&) = delete;
+
+    /**
+     * Copy assignment operator (deleted).
+     */
+    TriangleNodesProvisioner& operator=(const TriangleNodesProvisioner&) = delete;
+
+    /**
+     * Move constructor.
+     */
+    TriangleNodesProvisioner(TriangleNodesProvisioner&&) = default;
+
+    /**
+     * Move assignment operator.
+     */
+    TriangleNodesProvisioner& operator=(TriangleNodesProvisioner&&) = default;
+
+
+     /**
+      * Compute (x,y) nodes in equilateral triangle for polynomial of order N.
+      */
+      void computeEquilateralNodes(real_vector_type & x, real_vector_type & y) const;
+
+     /**
+      * Computes the Jacobian (determinant), the geometric factor rx (dr/dx), 
+      * and Fscale using nodes and differentiation matrix.
+      */
+      void computeJacobian();
+
+     /**
+      * Evaluates the 2D orthonormal polynomial on the simplex at (a,b) of order (i,j).
+      */
+      void evaluateSimplexPolynomial(const real_vector_type & a, const real_vector_type & b, const index_type i, const index_type j, real_vector_type & p) const;
+
+     /**
+      * Maps from (r,s) to (a,b) coordinates in the triangle.
+      */     
+      void rsToab(const real_vector_type & r, const real_vector_type & s, real_vector_type & a, real_vector_type & b) const;
+
+     /**
+      * Maps from (x,y) in the equilateral triangle to (r,s) coordinates in standard triangle
+      */
+      void xyTors(const real_vector_type & x, const real_vector_type & y, real_vector_type & r, real_vector_type & s) const;
+
+     /**
+      * Computes geometric warp function for interior nodes.
+      */
+      void computeWarpFactor(const real_vector_type & r, real_vector_type & warpFactor) const;
+
+     /**
+      * Computes the 2D Vandermonde matrix which maps modal coefficients to nodal values.
+      * @param[in] N Order of the approximating polynomials.
+      * @param[in] r 1st coordinate for standard triangle.
+      * @param[in] s 2nd coordinate for standard triangle.
+      * @param[out] V Vandermonde matrix.
+      */
+      void computeVandermondeMatrix(const int N, const real_vector_type & r, const real_vector_type & s, const real_matrix_type & V) const;
+
```

Comment:
Just being picky, but normally if you pass by value you don't bother with the const.

Reply:
Same as in evaluateSimplexPolynomial.